src/expr/src/relation/canonicalize.rs

```diff
@@ -15,8 +15,10 @@ use repr::{Datum, RelationType, ScalarType};
 
 /// Canonicalize equivalence classes of a join.
 ///
-/// This function makes it so that the same expression appears in only one
-/// equivalence class. It also sorts and dedups the equivalence classes.
+/// This function:
+/// * ensures the same expression appears in only one equivalence class.
+/// * ensures the equivalence classes are sorted and dedupped.
+/// * simplifies expressions to involve the least number of non-leaves.
```

Comment:
I don't understand what this last bullet point means. If it is important, could you say more?